newIDE/app/src/EventsFunctionsExtensionsLoader/MetadataDeclarationHelpers.js

```diff
@@ -910,37 +1140,69 @@ export const declareObjectPropertiesInstructionAndExpressions = (
-          '', // See below for adding the extra information
-          parameter.isOptional()
-        );
-        instructionOrExpression.setParameterLongDescription(
-          parameter.getLongDescription()
-        );
-        instructionOrExpression.setDefaultValue(parameter.getDefaultValue());
-      } else {
-        instructionOrExpression.addCodeOnlyParameter(
-          parameter.getType(),
-          '' // See below for adding the extra information
-        );
-      }
-      // Manually add the "extra info" without relying on addParameter (or addCodeOnlyParameter)
-      // as these methods are prefixing the value passed with the extension namespace (this
-      // was done to ease extension declarations when dealing with object).
-      instructionOrExpression
-        .getParameter(instructionOrExpression.getParametersCount() - 1)
-        .setExtraInfo(parameter.getExtraInfo());
+  const addParameter = (parameter: gdParameterMetadata) => {
+    if (!parameter.isCodeOnly()) {
+      instructionOrExpression.addParameter(
+        parameter.getType(),
+        parameter.getDescription(),
+        parameter.getExtraInfo(),
+        parameter.isOptional()
+      );
+      instructionOrExpression.setParameterLongDescription(
+        parameter.getLongDescription()
+      );
+      instructionOrExpression.setDefaultValue(parameter.getDefaultValue());
+    } else {
+      instructionOrExpression.addCodeOnlyParameter(
+        parameter.getType(),
+        parameter.getExtraInfo()
+      );
     }
-  );
+  };
+
+  const functionType = eventsFunction.getFunctionType();
+  const getterFunction = eventsFunctionsContainer.hasEventsFunctionNamed(
+    eventsFunction.getGetterName()
+  )
+    ? eventsFunctionsContainer.getEventsFunction(eventsFunction.getGetterName())
+    : null;
+  // This is used instead of getParametersForEvents because the Value parameter
+  // is already add by useStandardOperatorParameters.
+  const parameters = (functionType === gd.EventsFunction.ActionWithOperator &&
+  getterFunction
+    ? getterFunction
+    : eventsFunction
+  ).getParameters();
+
+  mapVector(parameters, addParameter, 0, userDefinedFirstParameterIndex);
+
+  if (functionType === gd.EventsFunction.ExpressionAndCondition) {
+    ((instructionOrExpression: any): gdMultipleInstructionMetadata).useStandardParameters(
+      'number'
+    );
+  } else if (functionType === gd.EventsFunction.StringExpressionAndCondition) {
+    ((instructionOrExpression: any): gdMultipleInstructionMetadata).useStandardParameters(
+      'string'
+    );
+  } else if (functionType === gd.EventsFunction.ActionWithOperator) {
+    ((instructionOrExpression: any): gdInstructionMetadata).useStandardOperatorParameters(
+      getterFunction &&
+        getterFunction.getFunctionType() ===
+          gd.EventsFunction.StringExpressionAndCondition
+        ? 'string'
+        : 'number'
+    );
+  }
+
+  mapVector(parameters, addParameter, userDefinedFirstParameterIndex);
```

Comment:
I'm a bit unease at this additional parameter which is "not standard". Could we instead remove it and add, in the  `addParameter`, a check on the index?

Reply:
mapVector, like map in JS, pass the index as the second parameter to the callback.

Reply:
I could create a local helper method that would filter the indexes between `mapVector` and `addParameter`.